documentation/modules/payload/android/meterpreter/injection.md

```diff
@@ -31,12 +31,13 @@ $ adb pull /data/app/com.existing.app-1/base.apk com.existing.apk
 APK Injection (as opposed to generating a single APK payload) requires a few tools
 to be present on your command line already:
 
+* [Java](https://www.oracle.com/java/technologies/downloads/) - Used for building/editing APK
 * [Apktool](https://ibotpeaches.github.io/Apktool/) - Used for rebuilding the APK
 * [keytool](https://docs.oracle.com/javase/8/docs/technotes/tools/unix/keytool.html) - To create and extract signing certificates
-* [jarsigner](https://docs.oracle.com/javase/7/docs/technotes/tools/windows/jarsigner.html) - To re-sign the APK
+* [apksigner](https://developer.android.com/studio/command-line/apksigner) - To re-sign the APK
 
 Installing these tools (if they are not installed already) will depend on your OS.
-Apktool can be installed manually or automatically (e.g `brew install apktool`).
+Apktool can be installed manually or automatically (e.g `brew install apktool` OR `apt install apktool`).
 keytool and jarsigner can be installed by installing the appropriate JDK.
```

Comment:
`jarsigner` is no longer required. `apksigner` is used instead.

`apksigner` can be installed on Linux with `apt install apksigner`. I'm not sure about OSX - `brew install apksigner` might work.

Reply:
Yes I updated injection.md by your given information.

Reply:
`keytool` is still required but the comment about installing the appropriate JDK has been removed.